src/v/net/conn_quota.cc

```diff
@@ -239,6 +248,12 @@ conn_quota::get_home_allowance(ss::net::inet_address addr) {
     assert_on_home(addr);
 
     if (addr == ss::net::inet_address{}) {
+#ifndef NDEBUG
```

Comment:
we don't have debug asserts. if we add an assert is because we need them. there are no debug asserts in any part of the codebase.

Reply:
this is a good assert to leave on prod mode as well. this would mean a severe logic bug and a memory leak.

Reply:
Ah - there is in the header of this file. I can remove, there is a log that will catch this anyways.

Reply:
look at iobuf for similar pattern.

Reply:
https://github.com/redpanda-data/redpanda/blob/34308282dc5c8e3293e33dda333692d2b9051395/src/v/net/conn_quota.h#L170-L174